### PR TITLE
DefinitionList story: convert knobs and actions to controls / args 

### LINKED ISSUE #13060: DefinitionList story: convert knobs and actions to controls / args
**Describe the bug**
As part of a push to convert all stories that use deprecated `@storybook/addon-knobs` and `@storybook/addon-actions` to use [controls](https://storybook.js.org/docs/react/essentials/controls) and [action argType annotation](https://storybook.js.org/docs/react/essentials/actions#action-argtype-annotation)

Convert `DefinitionList` story in `ui/components/ui/definition-list/definition-list.stories.js` to use [controls](https://storybook.js.org/docs/react/essentials/controls) and [action argType annotation](https://storybook.js.org/docs/react/essentials/actions#action-argtype-annotation)

Read the [MetaMask extension storybook documentation guidelines](https://metamask.github.io/metamask-storybook/?path=/story/getting-started-documentation-guidelines--page) to get a better understanding of writing stories. Providing MDX documentation although helpful, is a bonus and not a requirement for this issue.

To start storybook:
```
yarn storybook
```

**Requirements**
- [ ] Story has `argTypes` that align with component api / props
- [ ] All instances of `@storybook/addon-knobs` have been removed in favour of control args
- [ ] All instances of `@storybook/addon-actions` have been removed in favour of [action argType annotation](https://storybook.js.org/docs/react/essentials/actions#action-argtype-annotation)

Any further improvements to the story interaction or documentation is a bonus

**Images**

Screen Shot 2021-12-12 at 5 41 29 PM
Screen Shot 2021-12-12 at 5 41 47 PM

### PR DESCRIPTION
**Pull request by wiserdor**
## Explanation
Fixes #13060

deleted deprecated knobs and added argType with controllers and actions.

## Pre-Merge Checklist

- [ ] PR template is filled out
- [ ] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [ ] PR is linked to the appropriate GitHub issue
- [ ] PR has been added to the appropriate release Milestone

### + If there are functional changes:

- [ ] Manual testing complete & passed
- [ ] "Extension QA Board" label has been applied

### CODE REVIEW

File: ui/components/ui/definition-list/definition-list.stories.js [hunk 1]
```diff
@@ -34,39 +28,83 @@ const tooltips = {
...
+export default {
+  title: 'Components/UI/DefinitionList',
+  id: __filename,
+  argTypes: {
+    dictionary: { control: 'object', name: 'Dictionary' },
+    gapSize: {
+      control: 'select',
+      name: 'Gap Size',
+      options: SIZES,
+      defaultValue: SIZES.SM,
```
Comment (georgewrmarshall) [inline ui/components/ui/definition-list/definition-list.stories.js:40]: options arg type control should to be an array
```suggestion
      options: Object.values(SIZES),
      defaultValue: Object.values(SIZES.SM),
```

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Looking good! Just have a few type issues. options should be an array. Also some other proptype errors in the console 

Screen Shot 2022-07-12 at 4 53 21 PM

File: ui/components/ui/definition-list/definition-list.stories.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (wiserdor) [reply ui/components/ui/definition-list/definition-list.stories.js]: fixed and added argTypes for only relevant stories

**Review by georgewrmarshall — APPROVED** [review]
LGTM 💯 Thanks for the contribution @wiserdor 🙏

**Review by NidhiKJha — APPROVED** [review]

**Pull request merged**